hypershift-operator/controllers/nodepool/haproxy.go

```diff
@@ -62,6 +65,27 @@ func (r *NodePoolReconciler) reconcileHAProxyIgnitionConfig(ctx context.Context,
 	apiServerExternalPort := util.APIPortWithDefaultFromHostedCluster(hcluster, config.DefaultAPIServerPort)
 	if util.IsPrivateHC(hcluster) {
 		apiServerExternalAddress = fmt.Sprintf("api.%s.hypershift.local", hcluster.Name)
+	} else {
```

Comment:
can we have a unit tested getApiServerExternalAddress so we don't drop by accident again?

Reply:
https://github.com/openshift/hypershift/pull/1652